astroid/interpreter/_import/util.py

```diff
@@ -2,15 +2,32 @@
 # For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
 # Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
 
-try:
-    import pkg_resources
-except ImportError:
-    pkg_resources = None  # type: ignore[assignment]
+from importlib.util import _find_spec_from_path
 
 
-def is_namespace(modname):
-    return (
-        pkg_resources is not None
-        and hasattr(pkg_resources, "_namespace_packages")
-        and modname in pkg_resources._namespace_packages
-    )
+def is_namespace(modname: str) -> bool:
+    found_spec = None
+
+    # find_spec() attempts to import parent packages when given dotted paths.
+    # That's unacceptable here, so we fallback to _find_spec_from_path(), which does
+    # not, but requires instead that each single parent ('astroid', 'nodes', etc.)
+    # be specced from left to right.
+    working_modname = ""
+    last_parent = None
+    for component in modname.split("."):
+        if working_modname:
+            working_modname += "." + component
+        else:
+            # First component
+            working_modname = component
+        try:
+            found_spec = _find_spec_from_path(working_modname, last_parent)
```

Comment:
Is there any benefit to caching this?
Importlib usually is quite expensive, but perhaps this private function isn't.

Reply:
I'll take a look, it does seem we might call this repeatedly on the same part.

Reply:
[2697e10](https://github.com/PyCQA/astroid/pull/1536/commits/2697e10c07dc8c98fb3871ed4b7fce23e11d0460)